plan/task_profile.go

```diff
@@ -275,25 +320,24 @@ func (sel *Selection) attach2TaskProfile(profiles ...taskProfile) taskProfile {
 	return profile
 }
 
+func (sel *Selection) splitPushDownConditions() {
+	if len(sel.pushDownConditions)+len(sel.residualConditions) != 0 {
+		return
+	}
+	_, sel.pushDownConditions, sel.residualConditions = ExpressionsToPB(sel.ctx.GetSessionVars().StmtCtx, sel.Conditions, sel.ctx.GetClient())
+}
+
 // splitSelectionByIndexColumns splits the selection conditions by index schema. If some condition only contain the index
 // columns, it will be pushed to index plan.
-func (sel *Selection) splitSelectionByIndexColumns(schema *expression.Schema) (indexSel *Selection, tableSel *Selection) {
-	var tableConds []expression.Expression
-	var indexConds []expression.Expression
-	for _, cond := range sel.Conditions {
+func splitConditionsByIndexColumns(conditions []expression.Expression, schema *expression.Schema) (tableConds []expression.Expression, indexConds []expression.Expression) {
```

Comment:
Seems it play the same role as `DetachIndexFilterCondition` in `refiner.go`, why that one considered pkCol and this not?